### PR TITLE
Suggestion methods

### PR DESCRIPTION
**Pull request by phlax**
Moves suggestion methods from models class to SuggestionsReview class

next step is to tidy up a bit - and remove call to Unit.save() hopefully

### CODE REVIEW

File: pootle/apps/pootle_store/utils.py [hunk 1]
```diff
@@ -40,12 +49,159 @@ def add_comments(self, comment):
...
+
+        # FIXME: remove such a dependency on `ScoreLog`
+        # Update current unit instance's attributes
+        # important to set these attributes after saving Submission
+        # because in the `ScoreLog` we need to access the unit's certain
+        # attributes before it was saved
+        unit.submitted_by = suggestion_user
+        unit.submitted_on = current_time
+        unit.reviewed_by = self.reviewer
+        unit.reviewed_on = unit.submitted_on
```
Comment (unho) [inline pootle/apps/pootle_store/utils.py]: Nit: Use `current_time` too.

File: pootle/apps/pootle_store/utils.py [hunk 2]
```diff
@@ -40,12 +49,159 @@ def add_comments(self, comment):
...
+        # these do not conflict with `ScoreLog`'s interests, so it's safe
+        unit.target = suggestion.target
+        if unit.state == FUZZY:
+            unit.state = TRANSLATED
+
+        if suggestion.user_id is not None:
+            suggestion_user = suggestion.user
+        else:
+            User = get_user_model()
+            suggestion_user = User.objects.get_nobody_user()
```
Comment (phlax) [inline pootle/apps/pootle_store/utils.py]: im not changing any logic in this PR jut refactoring into the class

File: pootle/apps/pootle_store/utils.py (hunk 1 above)
Comment (phlax) [inline pootle/apps/pootle_store/utils.py]: see above comment

**Comment by unho** [comment]
lgtm

**Pull request merged**